sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GatewayAddressCache.java

```diff
@@ -135,6 +137,16 @@ public GatewayAddressCache(
              DefaultSuboptimalPartitionForceRefreshIntervalInSeconds);
     }
 
+
+    @Override
+    public void removeAddresses(final PartitionKeyRangeIdentity partitionKeyRangeIdentity) {
```

Comment:
Is remove un-conditional or conditional?

I.e. in-case of high throughut clients simple remove might result in removing existing valid refresh entries. 
Do they need to be based on the stale state instead?

Reply:
This is very much worth discussion. I'll give this one some thought.